lxd/main_forkexec.go

```diff
@@ -84,41 +109,41 @@ func cmdForkExec(args *Args) error {
 			}
 			env = append(env, arg)
 		} else if section == "cmd" {
-			cmd = append(cmd, arg)
+			command = append(command, arg)
 		} else {
-			return SubCommandErrorf(-1, "Invalid exec section: %s", section)
+			return fmt.Errorf("Invalid exec section: %s", section)
 		}
 	}
 
 	opts.Env = env
 
-	status, err := c.RunCommandNoWait(cmd, opts)
+	status, err := d.RunCommandNoWait(command, opts)
 	if err != nil {
-		return SubCommandErrorf(-1, "Failed running command: %q", err)
+		return fmt.Errorf("Failed running command: %q", err)
 	}
 	// Send the PID of the executing process.
 	w := os.NewFile(uintptr(3), "attachedPid")
 	defer w.Close()
 
 	err = json.NewEncoder(w).Encode(status)
 	if err != nil {
-		return SubCommandErrorf(-1, "Failed sending PID of executing command: %q", err)
+		return fmt.Errorf("Failed sending PID of executing command: %q", err)
 	}
 
 	var ws syscall.WaitStatus
 	wpid, err := syscall.Wait4(status, &ws, 0, nil)
 	if err != nil || wpid != status {
-		return SubCommandErrorf(-1, "Failed finding process: %q", err)
+		return fmt.Errorf("Failed finding process: %q", err)
 	}
 
 	if ws.Exited() {
-		return SubCommandErrorf(ws.ExitStatus(), "")
+		os.Exit(ws.ExitStatus())
 	}
 
 	if ws.Signaled() {
 		// 128 + n == Fatal error signal "n"
-		return SubCommandErrorf(128+int(ws.Signal()), "")
+		os.Exit(128 + int(ws.Signal()))
```

Comment:
Oh, is the ` + ` in Go legit? It used to automatically correct this to `+`.

Reply:
apparently so, I guess it behaves differently on a return